### ISSUE TITLE
Result is very dirty, any idea?

### ISSUE THREAD

**Issue by Dev-Damon** [sparkfun/SparkFun_MLX90640_Arduino_Example#25]
![Screenshot_20220216-165525_TMS32](https://user-images.githubusercontent.com/52990710/154224012-99344a24-fc74-4ed6-a357-9df4457727cd.jpg)
![Screenshot_20220216-165531_TMS32](https://user-images.githubusercontent.com/52990710/154224020-43de80b3-0959-4c3f-9d42-43e1f384b578.jpg)
![Screenshot_20220216-165537_TMS32](https://user-images.githubusercontent.com/52990710/154224024-b894874c-fcd0-44c7-bb6a-b4d272c9b778.jpg)

I'm using ESP8266. but I'm working on the official FreeRTOS SDK of esp8266.
I'm just wondering what's happening in my sensor,...

I'm sure there is no problem with i2c sensor.
I've checked status register and ctrl register every reading sensor. it should fine.
ctrl reg : 0x1901
- 2Hz
I've tried with 0.5Hz to 4Hz but it works like same.

Have you guys ever been to this situation? 

Thanks in advanced.

**Comment by LBuydens** [comment]
It looks like an image distortion one can get if the power decoupling capacitor is not present or not close enough to the sensor.

**Comment by Dev-Damon** [comment]
> It looks like an image distortion one can get if the power decoupling capacitor is not present or not close enough to the sensor.
------ edited
Thanks for answering.

I added capacitors so the pattern disappears. but still chess pattern remains.
I have a huge doubt in hardware, not the code and the sensors. but I don't really don't know what the problem is.


![image](https://user-images.githubusercontent.com/52990710/154650098-6c6cf3c0-7122-45d3-a51b-0565753b97c6.png)
![image](https://user-images.githubusercontent.com/52990710/154650584-bc9c1573-a995-4d4f-81bc-3015944a4d51.png)

**Comment by AsdyCorp** [comment]
> > It looks like an image distortion one can get if the power decoupling capacitor is not present or not close enough to the sensor.
> > ------ edited
> > Thanks for answering.
> 
> I added capacitors so the pattern disappears. but still chess pattern remains. I have a huge doubt in hardware, not the code and the sensors. but I don't really don't know what the problem is.
> 
> ![image](https://user-images.githubusercontent.com/52990710/154650098-6c6cf3c0-7122-45d3-a51b-0565753b97c6.png) ![image](https://user-images.githubusercontent.com/52990710/154650584-bc9c1573-a995-4d4f-81bc-3015944a4d51.png)

Did you solved it?

**Comment by Dev-Damon** [comment]
> Did you solved it?

No. I just figured out with after process.... that is the maximum performance of the sensor.
I checked sensor with official software of melexis,(called official manager of melexis) but it couldnt solved. just used median filter, and IIR filter.
